src/main/java/com/distributedsystems/distributedcache/consistency/ConsistencyImpl.java

```diff
@@ -1,9 +1,77 @@
 package com.distributedsystems.distributedcache.consistency;
 
+import com.distributedsystems.distributedcache.AppConfig;
+import com.distributedsystems.distributedcache.Utilities.ControllerConfigurations;
+import com.distributedsystems.distributedcache.Utilities.Utils;
 import com.distributedsystems.distributedcache.controller.Controller;
+import com.distributedsystems.distributedcache.totalorderedbroadcast.TotalOrderBroadcastServiceGrpc;
+import com.distributedsystems.distributedcache.totalorderedbroadcast.TotalOrderedBroadcast;
+import io.grpc.ManagedChannel;
+import io.grpc.ManagedChannelBuilder;
+import io.grpc.StatusRuntimeException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
 
-public interface ConsistencyImpl {
+@Component
+public class ConsistencyImpl implements ConsistencyImplInterface {
 
-    public Controller.ReadResponse read(Controller.ReadRequest request);
-    public Controller.WriteResponse write(Controller.WriteRequest request);
+    @Autowired
+    private ControllerConfigurations appConfig;
+
+    private static final Logger logger = LoggerFactory.getLogger(ConsistencyImpl.class);
+
+    /*
+    * Implements local read
+     */
+    @Override
+    public Controller.ReadResponse read(ConsistencyRequest request){
+        String value = Utils.readFromRedis(request.getKey());
+        return Controller.ReadResponse.newBuilder().setValue(value).setSuccess(true).build();
+    }
+
+    /*
+     * Implements broadcast write
+     */
+    @Override
+    public Controller.WriteResponse write(ConsistencyRequest request) {
+        TotalOrderedBroadcast.BroadcastMessage.Builder builder = TotalOrderedBroadcast.BroadcastMessage.newBuilder();
+        builder.setKey(request.getKey());
+        builder.setValue(request.getValue());
+        builder.setTypeOfRequest(TotalOrderedBroadcast.RequestType.PUT);
+        builder.setLamportClock(request.getLamportClock());
+        try {
+            TotalOrderBroadcastServiceGrpc.TotalOrderBroadcastServiceBlockingStub client = getTOBClient();
+            //TotalOrderedBroadcast.Empty response = client.sendBroadcastMessage(builder.build());
+        }catch (StatusRuntimeException e) {
+            logger.info(e.getMessage());
+            return Controller.WriteResponse.newBuilder().setSuccess(false).build();
+        }
+
+        BroadcastStatus status = new BroadcastStatus();
+        request.getPendingRequests().put(request.getLamportClock(), status);
+        checkStatus(status);
+        return Controller.WriteResponse.newBuilder().setSuccess(true).build();
+    }
+
+    protected TotalOrderBroadcastServiceGrpc.TotalOrderBroadcastServiceBlockingStub getTOBClient(){
+
+        String tobHost = appConfig.tobHost;
+        int tobPort = appConfig.tobPort;
```

Comment:
We should not create a blocking stub. Create an Async stub to here. Follow the example already in src/main/java/com/distributedsystems/distributedcache/DistributedcacheApplication.java file

Reply:
Aysnc stub is when we do not want our client to block after the API call ie, execution can proceed after the call. In this case I do not want that. I want to block until the broadcast replies with an "OK" message ie it has received the broadcast request. If this request submission fails controller should reply back failed to the user. 

Is there any specific reason as to why you are suggesting to use Async stub?

Reply:
As we discussed on the call, the TOB server needs to be async! It won't be handling sync calls.

Reply:
Yes, whatever we discussed still holds true. We need to understand the difference between architectural discussion vs code implementation. Yesterday what we discussed was from architectural point of view. Our handling of broadcast is asynchronous ie controller sends a request to broadcast that request is not immediately fulfilled and tob replied back with "OK" message. Now broadcast is happening in the background. Controller is NOT waiting on the initial request, it got a "OK" response.It will then proceed to wait for fulfillment. For this part I am using wait and signal.